hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/types/HazelcastObjectType.java

```diff
@@ -109,13 +118,32 @@ private static void generateFullTypeString(HazelcastObjectType type, StringBuild
         sb.append(')');
     }
 
+    private static void escape(StringBuilder sb, String value) {
+        for (char c : value.toCharArray()) {
+            if (c == '(' || c == ':' || c == ',' || c == ')') {
+                sb.append('\\');
+            }
+            sb.append(c);
+        }
+    }
+
     public void addField(Field field) {
-        fields.add(field);
+        fieldsBuilder.add(field);
```

Comment:
if the builder is null then the method will throw NPE, may be worth throwing something more obvious, for example list is already finalized exception

Reply:
[`8818039c`..`217f6dc0`][1]

[1]: https://github.com/hazelcast/hazelcast/compare/8818039c4cdcb25814b2d161a97c8032d229f536..217f6dc09634511c35cda5fc73dba20851abd051

Reply:
hmm, builder still can be null

Reply:
[`217f6dc0`..`d321903d`][1]

[1]: https://github.com/hazelcast/hazelcast/compare/217f6dc09634511c35cda5fc73dba20851abd051..d321903d51eada32b24f6cd3ea95939e82b5fe35